File: src/main/java/de/com/fdm/bot/commands/HttpStatusCommand.java
package de.com.fdm.bot.commands;


import de.com.fdm.bot.Command;
import de.com.fdm.bot.Parameters;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Component;

@Component
public class HttpStatusCommand implements Command {
    @Override
    public String execute(Parameters params) {
        if (params.getArgs().size() == 0) {
            return "No status code provided";
        }

        int statusCode;
        try {
            statusCode = Integer.parseInt(params.getArgs().get(0));
        } catch (NumberFormatException e) {
            return "Not a valid status code";
        }

        String statusCodeText = HttpStatus.valueOf(statusCode).toString();
        if (statusCodeText.contains("Unknown Status")) {
            return "Not a valid status code";
        }

        return String.format("%s https://http.cat/%s", statusCodeText, statusCode);
    }
}

File: src/main/java/de/com/fdm/bot/commands/HttpStatusCommand.kt
package de.com.fdm.bot.commands

import de.com.fdm.bot.Command
import de.com.fdm.bot.api.twitch.tmi.ReceiverMessage
import org.springframework.http.HttpStatus
import org.springframework.stereotype.Component

@Component
class HttpStatusCommand : Command {
    override fun execute(receiverMessage: ReceiverMessage): String {
        if (receiverMessage.getArgs().isEmpty()) {
            return "No status code provided"
        }

        val code: Int = try {
            receiverMessage.getArgs()[0].toInt()
        } catch (e: NumberFormatException) {
            return "Not a valid status code"
        }

        try {
            val statusCode = HttpStatus.valueOf(code).toString()
            return String.format("%s https://http.cat/%s", statusCode, statusCode)
        } catch (e: IllegalArgumentException) {
            return "Not a valid status code"
        }
    }
}